invenio_records_resources/services/records/schema.py

```diff
@@ -64,30 +68,23 @@ def __init__(self, service, schema):
         # TODO: Change constructor to accept a permission_policy_cls directly
         self._permission_policy_cls = service.config.permission_policy_cls
 
-    def _build_context(self, base_context):
-        context = {**base_context}
-        default_identity = context["identity"]  # identity required in context
+    def set_field_permission_check(self, context):
+        default_identity = context_identity.get()
```

Comment:
this one is wrong, it does not use the `context["identity"]` as the first option, `context_identity.get()` should be the fallback